### DISCUSSION TITLE
Implement VS Code's Refactor

### DISCUSSION THREAD

**Discussion by lgaborini** [posit-dev/positron#5577] [Feedback]
Are there any plans to implement this functionality in Positron?
I have loved it with Python under VS Code, but AFAIK there is no alternative right now in R, not even RStudio's basic file-based "Rename in Scope" (which was still quite useful).
(Not sure if it's a property of the IDE or the language server.)
Thanks!

**Comment by jmcphers** [comment]
Not in the next few months, but we'd definitely consider for a future release. Are there any particular refactoring gestures you miss? (e.g. Rename Symbol? Extract Function? etc.)

**Reply by peginatta** [reply]
Hello there! I would LOVE to have the RStudio "Extract Function" in positron. I don't know if there are any plans for this.

<img width="733" height="364" alt="image" src="https://github.com/user-attachments/assets/b615b096-4e18-40e5-9524-0b6ea3a53f02" />

**Reply by juliasilge** [reply]
Thanks for this feature request! 🙌 I've opened https://github.com/posit-dev/positron/issues/9590 to track and prioritize it.

**Reply by peginatta** [reply]
Thank you very much.

**Comment by lgaborini** [comment]
I'm not super familiar with those, but Rename Symbol was the one I used the most.

**Comment by AlbertoFabbri93** [comment]
Rename Symbol would be super useful!

**Comment by juliasilge** [comment]
We've got "Rename Symbol" tracked here, if you would like to 👍 or add additional info: https://github.com/posit-dev/positron/issues/4525
